### PR TITLE
Enabled LCM build on Windows 32bit.

### LINKED ISSUE #1715: try enabling lcm for 32-bit windows
@jhoare -- you said that lcm was working now on 32-bit windows (or specifically, that you didn't need the if 32-bit logic in the lcm CMakeLists.txt).  But lcm is not being build by the drake test suite on 32-bit because of the logic in the drake superbuild at https://github.com/RobotLocomotion/drake/blob/master/CMakeLists.txt#L38 .

If you pull the lcm option ON out of those conditionals and get it to build on win32 (really 32), then fantastic.  otherwise, we should probably put the win32 check back into the lcm CMakeLists.txt, too?  The best, of course, would be to successfully build lcm on win32.

### PR DESCRIPTION
**Pull request by jhoare**
Fixes #1715

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -35,10 +35,8 @@ option(REMOVE_UNUSED_EXTERNALS "enable this to remove those projects from disk"
 option(WITH_EIGEN		"required c++ matrix library.  only disable if you have it already."    ON)
 option(WITH_SWIGMAKE    "helper tools to build python & MATLAB wrappers for C++ libraries with Eigen" ON)
 if (WIN32)
-   if (CMAKE_SIZEOF_VOID_P EQUAL 8)  # gtk only compiles on win64 right now
 	option(WITH_LCM 		"interprocess communications protocol for visualizers, etc" ON)
```
Comment (RussTedrake) [inline CMakeLists.txt]: can you please move this line (and the matching line in the else) out of the conditional?  much better to have it only in one place.

**Comment by RussTedrake** [comment]
woohoo!  great news.  Looks like it worked (but that it also caught a segfault now that the win32 builds more tests... probably a byte alignment issue)

**Comment by RussTedrake** [comment]
@jamiesnape -- the "Merge build failed. No test results found" is a very misleading tag.  In fact, the tests ran and exactly one failed.  Can we get a more useful message there?

**Comment by RussTedrake** [comment]
@jhoare -- the failing test is almost certainly a place where we (probably me) have allocate an shared_ptr for an object using make_shared instead of using allocated_shared and Eigen's aligned allocator.  There are plenty of examples in the code.  For background reading, see http://eigen.tuxfamily.org/dox-devel/group__TopicStlContainers.html
The other solution, which we have adopted in many places in the code because of the ugliness/verbosity of the allocated_shared calls is to call shared_ptr<>(new Object()), and accept the extra memory allocation that happens.  (for reference, see notes in http://en.cppreference.com/w/cpp/memory/shared_ptr/make_shared)

@bradking -- this had fallen off my mental stack, but it will impact the way that you're type erasure design needs to work in the Optimization classes, too.

**Comment by jhoare** [comment]
travis-ci timed out with gcc/osx  on ``runPendulumLQR`` test. Is this an expected failure? These changes really shouldn't have affected OSX at all.

**Comment by RussTedrake** [comment]
sadly yes.  it's a known issue (I haven't added it to the known issues because travis isn't using the known issue logic).  looks like this is good to go.  thanks!

**Comment by jamiesnape** [comment]
@RussTedrake Sorry, there is nothing I do about that. I can change the message before "No test results found", but that is always appended automatically by the Jenkins plugin.

**Pull request merged**